### ISSUE TITLE
After switching to ESP-IDF 5.1: The mbc_master_destroy() function no longer works  (IDFGH-10779)

### ISSUE THREAD

**Issue by ThomasDQS** [espressif/esp-modbus#31]
Goal: 

The ESP32 is working as a master device. I want to open a TCP/IP socket to a slave device, read some registers and stop the modbus communication stack, destroy the interface & free all used objects by using the “mbc_master_destroy()” function. 

This flow is perfectly working with the ESP-IDF v5.0.1 on a ESP32-C3-DevKitM-1. 

Problem: 
I want to get the same code flow working on a ESP32-C6-DevKitC-1, which required an upgrade of the ESP-IDF to v5.1. 
Opening the connection to the slave and reading the desired registers is working perfectly, only the “mbc_master_destroy()” function is not working; some unexpected events got triggered resulting in running an endless loop until the IDLE task wachtdog got triggered. 

Logs:  
```
..[0;32mI (7893) MB_TCP_MASTER_PORT: Socket (#54)(192.168.2.242) created..[0m
.[0;32mI (7893) MB_TCP_MASTER_PORT: Connected 1 slaves, start polling....[0m
.[0;32mI (7943) MODBUS: temperature: 20.000000.[0m
.[0;32mI (7943) MODBUS: active MAX current: 0.000000.[0m
.[0;33mW (10943) MB_TCP_MASTER_PORT: Modbus port task couldn't exit gracefully within timeout -> abruptly deleting the task..[0m
.[0;31mE (10943) MB_PORT_COMMON: xMBMasterPortEventGet: Incorrect event triggered = 0..[0m
.[0;31mE (10943) MB_PORT_COMMON: eMBMasterPoll: Unexpected event triggered 0x00..[0m
.[0;31mE (15933) task_wdt: Task watchdog got triggered. The following tasks/users did not reset the watchdog in time:.[0m
.[0;31mE (15933) task_wdt:  - IDLE (CPU 0).[0m
.[0;31mE (15933) task_wdt: Tasks currently running:.[0m
.[0;31mE (15933) task_wdt: CPU 0: modbus_tcp_mast.[0m
.[0;31mE (15933) task_wdt: Print CPU 0 (current core) registers.[0m
Core  0 register dump:
MEPC    : 0x4080a8c0  RA      : 0x4080a8b6  SP      : 0x4082a270  GP      : 0x408129b0  
TP      : 0x40804af4  T0      : 0x40028192  T1      : 0x10000000  T2      : 0xffffffff  
S0/FP   : 0x00000080  S1      : 0x00000080  A0      : 0x00000080  A1      : 0x00000080  
A2      : 0x00000000  A3      : 0x00000004  A4      : 0x4081b000  A5      : 0x00000001  
A6      : 0x00000001  A7      : 0x0000000a  S2      : 0xffffffff  S3      : 0x40828f40  
S4      : 0x00000000  S5      : 0x00000000  S6      : 0x00000000  S7      : 0x00000000  
S8      : 0x00000000  S9      : 0x00000000  S10     : 0x00000000  S11     : 0x00000000  
T3      : 0x00000000  T4      : 0x00000000  T5      : 0x00000000  T6      : 0x00000000  
MSTATUS : 0x00000000  MTVEC   : 0x00000000  MCAUSE  : 0x00000000  MTVAL   : 0x00000000  
MHARTID : 0x00000000  
```

Any help on this would be great, thanks!

**Comment by alisitsyn** [comment]
Hello @ThomasDQS,

Thank you for this issue. Could you additionally set the log verbosity to DEBUG in kconfig, restart your project and attach log here? Please also attach your sdkconfig and map file in the build folder or a whole project. This will allow to find the exact reason for this issue and fix the destroy sequence for v5.1.

Thank you.

**Comment by ThomasDQS** [comment]
Hello @alisitsyn 

Hereby the debug_log, the sdkconfig file & the map file. 

Thank you. 
[debug_log.txt](https://github.com/espressif/esp-modbus/files/12261820/debug_log.txt)
[sdkconfig.txt](https://github.com/espressif/esp-modbus/files/12261826/sdkconfig.txt)
[modbus_tcp_master_map.txt](https://github.com/espressif/esp-modbus/files/12261830/modbus_tcp_master_map.txt)

**Comment by alisitsyn** [comment]
@ThomasDQS ,

Thank you for update. The destroy sequence of master should be revised. I will update it and test as soon as possible.

**Comment by ThomasDQS** [comment]
@alisitsyn 

I moved the `managed_components` folder and made the changes in the `mbc_tcp_master.c` file as you indicated. These changes have solved the problem! Thank you for the assistance!

**Comment by alisitsyn** [comment]
@ThomasDQS ,

Thank you for your issue and the feedback. The fix will go through the formal process and will be fixed ASAP.
Feel free to reopen the issue if you observe something later.

**Comment by alisitsyn** [comment]
Hi @ThomasDQS ,

The fix has been merged and released in component version v1.0.12. The issue is closed. Please check the update and feel free to reopen the issue if you still have similar issue.
